dask/array/tests/test_cupy_creation.py

```diff
@@ -167,3 +168,38 @@ def test_tri_like(xp, N, M, k, dtype, chunks):
     xp_a = xp.tri(*args, like=da.from_array(cupy.array(())))
 
     assert_eq(xp_a, cp_a)
+
+
+def test_to_backend_cupy():
+    # Test that `Array.to_backend` works as expected
+    with config.set({"array.backend": "numpy"}):
+
+        # Start with cupy-backed array
+        x = da.from_array(cupy.arange(11), chunks=(4,))
+        assert isinstance(x._meta, cupy.ndarray)
+
+        # Calling default `to_backend` should move
+        # backend to `numpy`
+        x_new = x.to_backend()
+        assert isinstance(x_new._meta, np.ndarray)
```

Comment:
>Since you're defining the backend at the top, wouldn't it make sense that any array manipulation (including da.from_array) would imply the backend?

The backend configuration (i.e. `config.set({"array.backend": "numpy"})`) means that `numpy` is the desired backend to use for **suported** Array-creation functions.  `from_array` is not a supported creation function yet, and once the collection is created, the `"array.backend"` is ignored (from there all dispatching is based on the **existing** type).

The fact that the `"array.backend"` configuration **only** applies to collection creation is a big motivation for `to_backend` - The proposed API provides a mechanism to "make sure" your data is using the desired backend (even if your workflow required you to create your collection with a function that is **not** supported by your desired backend).

Reply:
Got it, thanks for the clarification!